docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md

```diff
@@ -0,0 +1,482 @@
+{
+    [WebMethod]
+    public string MyWebMethod(DataTable dataTable)
+    {
+        /* Web method implementation. */
+    }
+}
+```
+
+A variation on this is not to accept `DataSet` or `DataTable` directly as a parameter, but instead to accept it as part of the overall SOAP serialized object graph, as shown in the following code:
+
+```cs
+using System.Data;
+using System.Web.Services;
+
+[WebService(Namespace = "http://contoso.com/")]
+public class MyService : WebService
+{
+    [WebMethod]
+    public string MyWebMethod(MyClass data)
+    {
+        /* Web method implementation. */
+    }
+}
+
+public class MyClass
+{
+    // Property of type DataTable, automatically serialized and
+    // deserialized as part of the overall MyClass payload.
+    public DataTable MyDataTable { get; set; }
+}
+```
+
+Or, using WCF instead of ASP.NET web services:
+
+```cs
+using System.Data;
+using System.ServiceModel;
+
+[ServiceContract(Namespace = "http://contoso.com/")]
+public interface IMyContract
+{
+    [OperationContract]
+    string MyMethod(DataTable dataTable);
+    [OperationContract]
+    string MyOtherMethod(MyClass data);
+}
+
+public class MyClass
+{
+    // Property of type DataTable, automatically serialized and
+    // deserialized as part of the overall MyClass payload.
+    public DataTable MyDataTable { get; set; }
+}
+```
+
+In all of these cases, the threat model and security guarantees are the same as the [DataSet.ReadXml and DataTable.ReadXml section](#dsrdtr).
+
+## Deserialize a DataSet or DataTable via XmlSerializer
+
+Developers can use `XmlSerializer` to deserialize `DataSet` and `DataTable` instances, as shown in the following code:
+
+```cs
+using System.Data;
+using System.IO;
+using System.Xml.Serialization;
+
+public DataSet PerformDeserialization1(Stream stream) {
+    XmlSerializer serializer = new XmlSerializer(typeof(DataSet));
+    return (DataSet)serializer.Deserialize(stream);
+}
+
+public MyClass PerformDeserialization2(Stream stream) {
+    XmlSerializer serializer = new XmlSerializer(typeof(MyClass));
+    return (MyClass)serializer.Deserialize(stream);
+}
+
+public class MyClass
+{
+    // Property of type DataTable, automatically serialized and
+    // deserialized as part of the overall MyClass payload.
+    public DataTable MyDataTable { get; set; }
+}
+```
+
+In these cases, the threat model and security guarantees are the same as the [DataSet.ReadXml and DataTable.ReadXml section](#dsrdtr)
+
+## Deserialize a DataSet or DataTable via JsonConvert
+
+The popular third-party Newtonsoft library [JSON.NET](https://www.newtonsoft.com/json) can be used to deserialize `DataSet` and `DataTable` instances, as shown in the following code:
+
+```cs
+using System.Data;
+using Newtonsoft.Json;
+
+public DataSet PerformDeserialization1(string json) {
+    return JsonConvert.DeserializeObect<DataSet>(data);
+}
+
+public MyClass PerformDeserialization2(string json) {
+    return JsonConvert.DeserializeObect<MyClass>(data);
```

Comment:
@gewarren Correct me if I'm wrong, but isn't non default `JsonSerializerSettings` passed to the call an important prerequisite for the vulnerability? Otherwise JSON.NET won't deserialize types information.

Reply:
Tagging @GrabYourPitchforks who wrote the article.

Reply:
@JarLob that's what [the article states](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/dataset-datatable-dataview/security-guidance#deserialize-a-dataset-or-datatable-via-jsonconvert)

The popular third-party Newtonsoft library JSON.NET can be used to deserialize DataSet and DataTable instances, as shown in the following code:

This PR is merged, if you feel there's a problem please open an issue on the [document](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/dataset-datatable-dataview/security-guidance#deserialize-a-dataset-or-datatable-via-jsonconvert)

Reply:
Thanks for the link, will investigate it. I've commented here because didn't find how to comment on that document.

Reply:
The text of the doc is correct with respect to this point. `JSON.NET` (and most other ostensibly "safe" serializers) see `DataSet` from the _static_ type information, either because the caller is explicitly asking for `DataSet` to be deserialized or because it's requesting a type who has a `DataSet` property be deserialized. This will cause `JSON.NET` to activate its `DataSet` deserialization logic, which can lead to denial of service.